diff --git a/tabbed.c b/tabbed.c
index c05d7bf..d8ac459 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -326,6 +326,7 @@ drawbar(void)
 	XftColor *col;
 	int c, cc, fc, width, nbh, i;
 	char *name = NULL;
+  char tabtitle[512];
 
 	if (nclients == 0) {
 		dc.x = 0;
@@ -376,7 +377,9 @@ drawbar(void)
 		} else {
 			col = clients[c]->urgent ? dc.urg : dc.norm;
 		}
-		drawtext(clients[c]->name, col);
+    snprintf(tabtitle, sizeof(tabtitle), "%d: %s",
+        c + 1, clients[c]->name);
+		drawtext(tabtitle, col);
 		dc.x += dc.w;
 		clients[c]->tabx = dc.x;
 	}
